rc/lsp.kak

```diff
@@ -967,6 +955,15 @@ method    = "textDocument/semanticTokens/full"
 ' "${kak_session}" "${kak_buffile}" "${kak_opt_filetype}" "${kak_timestamp}" | eval ${kak_opt_lsp_cmd} --request) > /dev/null 2>&1 < /dev/null & }
 }
 
+define-command -hidden lsp-semantic-tokens-refresh %{
+    evaluate-commands %sh{
+      eval "set -- $kak_quoted_buflist"
+      for buffer in "$@"; do
+        echo eval -buffer "$buffer" lsp-semantic-tokens-request
```

Comment:
> We'll ask for new tokens anyway via the NormalIdle hook, right?

This works now, but I want to get rid of using NormalIdle for this and have a global option that sends `lsp-semantic-tokens` only when the file changes. If that happens, then there'd be stale highlighting for background buffers when there's a refresh notification.

When I was reading the LSP spec, I was under the impression that the refresh request would be called relatively infrequently (like if the project just loaded, or the project configuration changed) so it shouldn't necessarily need this optimization yet - how often does rust-analyzer send it?

Reply:
rust-analyzer seems to send `workspace/semanticTokens/refresh` whenever the content changes (after `textDocument/didChange`). I also thought that this was just in case the project config changed.

Maybe we should go with something like

```diff
diff --git a/rc/lsp.kak b/rc/lsp.kak
index 9ad2743..8aa082e 100644
--- a/rc/lsp.kak
+++ b/rc/lsp.kak
@@ -239,3 +239,6 @@ draft    = """
 ' "${kak_session}" "${kak_buffile}" "${kak_opt_filetype}" "${kak_timestamp}" "${lsp_draft}" | eval ${kak_opt_lsp_cmd} --request
-printf %s\\n "${kak_opt_lsp_callback}" | kak -p "${kak_session}"
+{
+    printf %s\\n "${kak_opt_lsp_callback}"
+    echo lsp-semantic-tokens-request # TODO only if enabled
+} | kak -p "${kak_session}"
 ) > /dev/null 2>&1 < /dev/null & }
```

Though I have tried that before fore code actions, and it failed miserably (I got stale code actions, or none at all)

Not sure why.

Reply:
merged most commits except for the last one, since I'm not sure if the refresh support is alreayd useful? I think it will be when we can turn off the NormalIdle hook